src/main.rs

```diff
@@ -1,35 +1,335 @@
-        write!(out, "{}{}", sample, if i % channels as usize == (channels - 1).into() { "\n" } else { " " }).unwrap();
+    for cur_data_index in 0..data_length
+    {
+        for cur_channel_index in 0..channels
+        {
+            let mut cur_data = 0.0;
+            unsafe
+            {
+                cur_data = data_array[cur_channel_index][cur_data_index];
+            }
+            if cur_channel_index == channels - 1
+            {
+                writeln!(out, "{}", cur_data).unwrap();
+            }
+            else 
+            {
+                write!(out, "{} ", cur_data).unwrap();
+            }
+        }
+    }
+    
+    
+}
+fn process_data_array(block_size : usize, channels : usize, comb_filter:&mut CombFilter, data_length:usize)
+{
+    
+    for cur_channel in 0..channels
+    {
+        let mut cur_block = 0;
+        loop {
+            let mut finished = false;
+            let start_index = cur_block * block_size;
+            let mut block_data : [f32; max_block_size] = [0.0; max_block_size];
+            for data_index in 0 .. block_size
+            {
+                let cur_index = start_index + data_index;
+                unsafe{
+                    if cur_index < data_length
+                    {
+                        block_data[data_index] = data_array[cur_channel][cur_index];
+                    }
+                    else 
+                    {
+                        finished = true;
+                    }
+                }
+            } 
+            let mut processed_data : [f32; max_block_size] = [0.0; max_block_size];
+            let mut block_data_single_channel = [& block_data[0..block_size]];
+            let mut processed_data_single_channel = [&mut processed_data[0..block_size]];
+            //comb_filter.process_block(&mut block_data, &mut processed_data, block_size);
+            comb_filter.process(& block_data_single_channel[0..1], &mut processed_data_single_channel[0..1]);
+            for data_index in 0 .. block_size
+            {
+                let cur_index = start_index + data_index;
+                unsafe{
+                    if cur_index < data_length
+                    {
+                        data_array[cur_channel][cur_index] = processed_data[data_index];
+                    }
+                    else 
+                    {
+                        break;
+                    }
+                }
+            } 
+            if finished
+            {
+                break;
+            }
+            cur_block = cur_block + 1;
+            
+        }
+
+    }
+}
+fn generate_sine_wave(num_samples: usize, frequency: f32, amplitude: f32, sample_rate: f32)  
+{
+    use std::f32::consts::PI;
+    for cur_index in 0..num_samples
+    {
+        let cur_number = amplitude * (2.0 * PI * frequency * cur_index as f32 / sample_rate).sin();
+        unsafe
+        {
+            data_array[0][cur_index] = cur_number;
```

Comment:
I would try making this test dynamic here for multiple channels. Maybe even have different frequencies on each channel.